omp/matrix/dense_kernels.cpp

```diff
@@ -455,6 +456,40 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_DENSE_CONVERT_TO_SELLP_KERNEL);
 
 
+template <typename ValueType, typename IndexType>
+void convert_to_sparsity(std::shared_ptr<const OmpExecutor> exec,
+                         matrix::Sparsity<ValueType, IndexType> *result,
+                         const matrix::Dense<ValueType> *source)
+{
+    auto num_rows = result->get_size()[0];
+    auto num_cols = result->get_size()[1];
+    auto num_nonzeros = result->get_num_nonzeros();
+
+    auto row_ptrs = result->get_row_ptrs();
+    auto col_idxs = result->get_col_idxs();
+
+    size_type cur_ptr = 0;
+    row_ptrs[0] = cur_ptr;
+    for (size_type row = 0; row < num_rows; ++row) {
+#pragma omp parallel for
+        for (size_type col = 0; col < num_cols; ++col) {
+            auto val = source->at(row, col);
+            if (val != zero<ValueType>()) {
+#pragma omp critical
+                {
+                    col_idxs[cur_ptr] = col;
+                    ++cur_ptr;
+                }
```

Comment:
Do you mean if the `col_idxs` need to be sorted ? I think that is fine, because it is not guaranteed and probably not required. Algorithms that require it implement their own checks for example, like the ParILU factorization does.

Reply:
We do not guarantee that CSR is sorted by column indices ([see CSR documentation](https://github.com/ginkgo-project/ginkgo/blob/develop/include/ginkgo/core/matrix/csr.hpp#L68)). For COO, we only guarantee that it is sorted by row indices ([COO documentation](https://github.com/ginkgo-project/ginkgo/blob/develop/include/ginkgo/core/matrix/coo.hpp#L63)).

Reply:
got it. thanks